src/sdk/k4a.c

```diff
@@ -526,171 +526,252 @@ static k4a_result_t validate_configuration(k4a_context_t *device, const k4a_devi
+            {
+                result = K4A_RESULT_FAILED;
+                LOG_ERROR("The configured subordinate device delay of %dusec from master device is not supported.",
+                          config->subordinate_delay_off_master_usec);
+            }
         }
-        else
+
+        if (config->wired_sync_mode != K4A_WIRED_SYNC_MODE_SUBORDINATE &&
+            config->subordinate_delay_off_master_usec != 0)
         {
-            result = K4A_RESULT_FROM_BOOL(config->subordinate_delay_off_master_usec == 0);
+            result = K4A_RESULT_FAILED;
+            LOG_ERROR("The configured subordinate device delay of %dusec is not supported with %d wired sync mode.",
+                      config->subordinate_delay_off_master_usec,
+                      config->wired_sync_mode);
         }
     }
 
-    if (config->depth_mode != K4A_DEPTH_MODE_OFF)
+    if (K4A_SUCCEEDED(result))
     {
-        depth_enabled = true;
-    }
+        if (config->depth_mode != K4A_DEPTH_MODE_OFF)
+        {
+            depth_enabled = true;
+        }
 
-    if (config->color_resolution != K4A_COLOR_RESOLUTION_OFF)
-    {
-        color_enabled = true;
-    }
+        if (config->color_resolution != K4A_COLOR_RESOLUTION_OFF)
+        {
+            color_enabled = true;
+        }
 
-    if (depth_enabled && color_enabled)
-    {
-        if (K4A_SUCCEEDED(result))
+        if (depth_enabled && color_enabled)
         {
             int64_t fps = 1000000 / k4a_convert_fps_to_uint(config->camera_fps);
-            result = K4A_RESULT_FROM_BOOL((config->depth_delay_off_color_usec >= -fps) &&
-                                          (config->depth_delay_off_color_usec <= fps));
+            if (config->depth_delay_off_color_usec < -fps || config->depth_delay_off_color_usec > fps)
+            {
+                result = K4A_RESULT_FAILED;
+                LOG_ERROR(
+                    "The configured delay of %dusec between depth capture and color image capture is not supported.",
+                    config->depth_delay_off_color_usec);
+            }
         }
-    }
-    else
-    {
-        if (K4A_SUCCEEDED(result))
+        else if (!depth_enabled && !color_enabled)
         {
-            result = K4A_RESULT_FROM_BOOL(config->depth_delay_off_color_usec == 0);
+            result = K4A_RESULT_FAILED;
+            LOG_ERROR("Neither depth camera nor color camera are enabled in the configuration, at least one needs to "
+                      "be enabled.",
+                      0);
         }
-
-        if (K4A_SUCCEEDED(result))
+        else
         {
-            result = K4A_RESULT_FROM_BOOL(config->synchronized_images_only == false);
+            if (config->depth_delay_off_color_usec != 0 || config->synchronized_images_only)
+            {
+                result = K4A_RESULT_FAILED;
+                LOG_ERROR("Both depth camera and color camera must be enabled to modify configuration of depth delay "
```

Comment:
Probably should split this in to two errors as well.
LOG_ERROR("Both depth camera and color camera must be enabled to set depth_delay_off_color_usec");
LOG_ERROR("Both depth camera and color camera must be enabled to enable synchronized_images_only");